From 291a374bd281b63be4b8d4f5667c0a7ae2e79019 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 3 May 2024 11:49:01 +0200
Subject: [PATCH] irqchip: rpmsg: fix may be used uninitialized warning

The 'ret' variable is used uninitialized in dev_err.
Store in 'ret' variable the irq_rpmsg_read_val() return to fix
the issue.

Change-Id: I9622dcdadd231c334a056af42904680289a83b1e
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/378202
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 drivers/irqchip/irq-rpmsg.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-rpmsg.c b/drivers/irqchip/irq-rpmsg.c
index 37db2ddfca50..f38d5aa2ace3 100644
--- a/drivers/irqchip/irq-rpmsg.c
+++ b/drivers/irqchip/irq-rpmsg.c
@@ -164,7 +164,8 @@ static void irq_rpmsg_setup_work(struct work_struct *ws)
 	struct rpmsg_intc_msg msg = {VIRT_INTC_CFG, 0, 0};
 	int ret;
 
-	if (irq_rpmsg_read_val(rirq_dev, &msg)) {
+	ret = irq_rpmsg_read_val(rirq_dev, &msg);
+	if (ret) {
 		dev_err(&rpdev->dev, "read intc cfg failed: %d\n", ret);
 		goto err;
 	}
-- 
2.39.5

